lib/graphql/schema/loader.rb

```diff
@@ -102,7 +102,7 @@ def define_type(type, type_resolver)
               name: type["name"],
               type: type_resolver.call(type["type"]),
               description: type["description"],
-              default_value: type["defaultValue"]
+              default_value: type["defaultValue"] ? JSON.parse(type["defaultValue"], quirks_mode: true) : nil
```

Comment:
Dove-tailing off my comment above about the enum loader fix, the application values for enum's is one of those things that gets lost in introspection (just like field resolvers).

At the time that the schema is being reloaded, the data being passed in from `defaultValue` is _always_ going to contain the GraphQL values (because we call `coerce_result` when it's dumped).

My thinking was:
- For enum's, we're OK, because by default, the GraphQL and application values are the same. (And per my comment above, we don't know the application values anyway)
- For the other built-in types, like Int and Float, we're also OK, because `JSON.parse` will give you the right native Ruby types.
- For custom scalars, you'll get whatever the external representation was (potentially any valid JSON value). We probably can't do much better than that, since we don't have the `coerce_input` proc's defined.

So back to your comment about testing, I think where that particular kind of test breaks down is:
- Define an enum with a custom value
- Export the schema via introspection; the custom value is lost
- Reload the schema, the best we can do is make the internal and external values the same
- Assert default value equality, and :boom:

I think the best we can test for is:
- Given different types of default values (enum's, input object types, lists, etc), it doesn't break.
- For custom scalars, the default value is equal to whatever you got from `coerce_result` on the original value.
- If you import a schema, you can export it again without any problems. (This test would probably break right now, since export relies on `coerce_result`, which is left undefined by the loader).

Hopefully that long train of thought made sense. I'll add some more tests for those cases I mentioned. Your thoughts?